### PR TITLE
TGP-2678/update homepage

### PR DESCRIPTION
**Pull request by Reece-Carruthers**
# Checklist for developers before reviewing
 - [x] I have merged in main
 - [x] I have assigned the PR to me
 - [x] I ran the code with a clean compile test and ensured there are no warnings or deprecations
 - [x] I have run `sbt prePR` to run all tests, format and check the code coverage
 - [x] I have written unit tests to cover any code that I wrote, where necessary
 - [x] I have linked tickets to any TODOs I have written
 - [x] I have run the microservices with the latest SM2 config
 - [x] I have checked all the Acceptance Criteria pass
 - [x] I have run the pipeline journey tests
 - [x] I have run the local journey tests
 - [x] I have spoken to the QAs about any changes that may need to be made to the journey tests OR no changes to the journey tests are necessary
 - [x] I have moved the ticket to Review in Jira
 - [x] I have left a message in tgp-prs on slack to alert other Devs that this ticket is ready for review
 - [x] I have checked my PR is not in draft mode

# Checklist for reviewer 1
- [ ] I have added an emoji to the relevant message in tgp-prs to show I am reviewing
- [ ] I have checked out the branch
- [ ] I have run the microservices with the latest SM2 config
- [ ] I have checked all the Acceptance Criteria pass
- [ ] I reviewed the code and left comments where necessary
- [ ] I have left a message on the relevant tgp-prs message to show that I have finished reviewing

# Checklist for reviewer 2
- [ ] I have added an emoji to the relevant message in tgp-prs to show I am reviewing
- [ ] I have checked out the branch
- [ ] I have run the microservices with the latest SM2 config
- [ ] I have checked all the Acceptance Criteria pass
- [ ] I reviewed the code and left comments where necessary
- [ ] I have left a message on the relevant tgp-prs message to show that I have finished reviewing

# Checklist for developers before merging
 - [x] The PR has been approved by 2 other developers
 - [x] I have merged in main
 - [x] I have run `sbt preMerge` to run all tests, check formatting and code coverage
 - [x] I have run the microservices with the latest SM2 config
 - [x] I have checked all the Acceptance Criteria pass
 - [x] I have checked that the QAs have merged in any changes to the journey tests
 - [x] I have run the pipeline journey tests
 - [x] I have run the local journey tests

# Reminders for developers after merging
 1. Wait for pipeline to pass
 2. Move ticket to QA column in Jira
 3. Leave message in tgp-qa on slack to alert QAs that this ticket is ready for testing

### CODE REVIEW

**Comment by Reece-Carruthers** [comment]
image
image
image

File: app/views/GoodsRecordsView.scala.html [hunk 1]
```diff
@@ -165,9 +163,5 @@ <h1 class="govuk-heading-l">@messages("goodsRecords.title")</h1>
 
         <p class="govuk-body"><a href="@routes.HomePageController.onPageLoad().url" class="govuk-link">@messages("site.goBackToHomePage")</a></p>
 
-        @if(appConfig.downloadFileEnabled) {
```
Comment (Ewan-James-Donovan) [inline app/views/GoodsRecordsView.scala.html:168]: AC 1 ✅: Remove download TGP data link behaviour on /goods-profile

File: conf/messages.en [hunk 2]
```diff
@@ -202,11 +202,16 @@ homepage.p3.linkText = Create a new goods record
 homepage.h2.2 = Manage your profile
 homepage.p4.linkText = Ask for an opinion from HMRC if a commodity code is correct
 homepage.p5.linkText = Manage your profile details
-homepage.h2.3 = Help and contact
+homepage.h2.3 = Download TGP data
```
Comment (Ewan-James-Donovan) [inline conf/messages.en:205]: AC 2.1 ✅: H2 Download TGP data as per design

File: conf/messages.en [hunk 3]
```diff
@@ -202,11 +202,16 @@ homepage.p3.linkText = Create a new goods record
...
+homepage.h2.4 = Help and contact
 homepage.p6.linkText = Find additional support and contacts to help you
 homepage.downloadBanner.important = Important
 homepage.downloadBanner.content = The TGP records file you asked for is ready to download.
 homepage.downloadBanner.linkText = Go to the TGP records file page to download the file
+homepage.downloadLinkText.requestFile = Request a file of your TGP records
```
Comment (Ewan-James-Donovan) [inline conf/messages.en:211]: AC 2.2: Messages added, Get your TGP records file ( See attached for design), You do not have TGP goods records to download. ( See attached for design) etc

File: conf/messages.en [hunk 4]
```diff
@@ -407,9 +412,6 @@ goodsRecords.li2 = 'Goods description'
 goodsRecords.p3 = You do not have any records in your records list.
 goodsRecords.label = Search by trader reference, goods description or commodity code
 goodsRecords.link1 = Create a new goods record
-goodsRecords.downloadLinkText.requestFile = Request a file of your TGP records
```
Comment (Ewan-James-Donovan) [inline conf/messages.en:410]: AC 1: Messages removed, Remove download TGP data link behaviour on /goods-profile

File: app/views/HomePageView.scala.html [hunk 5]
```diff
@@ -58,7 +59,16 @@ <h2 class="govuk-heading-m">@messages("homepage.h2.2")</h2>
         <a href="@routes.ProfileController.onPageLoad().url" class="govuk-link">@messages("homepage.p5.linkText")</a>
     </p>
 
-    <h2 class="govuk-heading-m">@messages("homepage.h2.3")</h2>
+    @if(appConfig.downloadFileEnabled) {
+        <h2 class="govuk-heading-m">@messages("homepage.h2.3")</h2>
+        @if(downloadLinkMessagesKey == "homepage.downloadLinkText.noGoodsRecords") {
```
Comment (Ewan-James-Donovan) [inline app/views/HomePageView.scala.html:64]: AC 2.2 & 2.3 ✅: If user has goods records in their profile display hyperlink, If user does not have goods records in their profile display

File: app/controllers/HomePageController.scala [hunk 6]
```diff
@@ -32,15 +33,41 @@ class HomePageController @Inject() (
...
+      val downloadLinkMessagesKey = getDownloadLinkMessagesKey(downloadDataSummary, doesGoodsRecordExist)
+      downloadDataSummary match {
+        case Some(downloadDataSummary) if downloadDataSummary.status == FileReadyUnseen =>
+          Ok(view(downloadReady = true, downloadLinkMessagesKey))
+        case _                                                                          =>
+          Ok(view(downloadReady = false, downloadLinkMessagesKey))
+      }
     }
   }
+
```
Comment (Ewan-James-Donovan) [inline app/controllers/HomePageController.scala:57]: AC 2.2 & 2.3 ✅: If user has goods records in their profile display hyperlink, If user does not have goods records in their profile display -> The above and below logic seems fine

**Review by Ewan-James-Donovan — APPROVED** [review]
LGTM: All requirements covered :)

**Review by kenneth-rayner — APPROVED** [review]

**Pull request merged**